### DISCUSSION TITLE
怎么永久关闭自动连播？

### DISCUSSION THREAD

**Discussion by SunJunge08** [the1812/Bilibili-Evolved#3293] [提问 / Q&A]
如题，现在逼站为了增加播放量，不会记忆自动连播的开关情况。这就导致关闭自动连播后，再去看下一个视频，自动连播还是开着的。有没有办法可以永久关闭此功能？
有的时候想要在评论区说点事，刚打到一半，结果视频播放完毕直接自动播放下一个视频，自己打的评论全都没了……
真不知道发明这个功能的人是多少年脑血栓。

**Comment by SunJunge08** [comment]
顶一下 怎么没人回复

**Reply by the1812** [reply]
传统连播模式？

**Reply by SunJunge08** [reply]
> 传统连播模式？

我不太知道传统连播模式是哪一种，麻烦解释一下。

我要说的是这种：
![image](https://user-images.githubusercontent.com/70008772/180643735-afcc8b49-5c38-4886-8b20-b1b742bf5a90.png)

而不是这种：
![image](https://user-images.githubusercontent.com/70008772/180643728-fb3d4bd0-8c95-4fad-bdfb-3e3eede94ea0.png)

后者有记忆功能，关闭之后不会自动开启；而前者没有记忆功能，每次打开这类“合集”视频都会自动开启，我还得手动关闭，要不然播放完它就自己放下一个视频，也就是我问题里写的那种情况。

**Reply by the1812** [reply]
传统连播模式就是合集 / 分P / 列表连播, 其他推荐视频不连播

不过这么看来也不符合你的要求

**Reply by SunJunge08** [reply]
> 传统连播模式就是合集 / 分P / 列表连播, 其他推荐视频不连播
> 
> 不过这么看来也不符合你的要求

那我的问题怎么解决，我就想把所有的连播按钮都自动关掉

**Reply by the1812** [reply]
没有现成的功能，等我优化下传统连播模式应该可以

**Reply by SunJunge08** [reply]
> 没有现成的功能，等我优化下传统连播模式应该可以

好的，期待大佬的新功能

**Reply by Geeyun-JY3** [reply]
实测了一下你说的前者的「自动连播」，好像能记忆吧，毕竟也是「更多播放设置 > 播放方式」里的一个设置项，要记忆不了就是 bug 了

**Reply by SunJunge08** [reply]
> 实测了一下你说的前者的「自动连播」，好像能记忆吧，毕竟也是「更多播放设置 > 播放方式」里的一个设置项，要记忆不了就是 bug 了

可我的记忆不了，关闭后刷新网页还是会自动打开（

**Reply by Cyiming** [reply]
> 实测了一下你说的前者的「自动连播」，好像能记忆吧，毕竟也是「更多播放设置 > 播放方式」里的一个设置项，要记忆不了就是 bug 了

设置只当天有效

**Reply by SunJunge08** [reply]
> > 实测了一下你说的前者的「自动连播」，好像能记忆吧，毕竟也是「更多播放设置 > 播放方式」里的一个设置项，要记忆不了就是 bug 了
> 
> 设置只当天有效

我的是仅限同一页面有效，刷新一下就有自动打开了（

**Comment by SunJunge08** [comment]
再顶一下，顺便补充一点：
如果我关掉 **Bilibili-Evolved** 这个脚本，是可以记忆的。但是打开脚本后，就会自动把“自动连播”的开关打开。

所以应该是脚本的问题，希望作者可以修改一下。

**Reply by Geeyun-JY3** [reply]
根据 #3705 的反馈，合集的自动连播记忆应该还是能用的，建议尝试找找是 Bilibili Evolved 哪个组件的问题

**Reply by SunJunge08** [reply]
> 根据 #3705 的反馈，合集的自动连播记忆应该还是能用的，建议尝试找找是 Bilibili-Evolved 哪个组件的问题

#3705 只说“播完暂停”和“自动连播”是联动的，没说它能记忆啊。
我在脚本的设置栏里只找到一个关于自动连播的选项，就是“传统连播模式”，但是它只管分P视频和番剧，而不是合集视频。
![pic](https://user-images.githubusercontent.com/70008772/193431325-52890237-b965-451b-a91f-d8d5764945ba.png)

**Reply by Geeyun-JY3** [reply]
我说的记忆是指破站本身会记忆「自动连播」是开还是关（即如你所说的「关掉 Bilibili Evolved 这个脚本，是可以记忆的」），所以建议你排查一下是不是 Bilibili Evolved 哪个组件造成的「自动把 `自动连播` 的开关打开」，比如 `传统连播模式` 就会影响到这个记忆。

**Reply by the1812** [reply]
目前这个开启应该是会让合集也连播的, 我正在考虑弄个开关, 因为合集有些是分 P 但有些又是无关视频

**Reply by Geeyun-JY3** [reply]
> 目前这个开启应该是会让合集也连播的, 我正在考虑弄个开关, 因为合集有些是分 P 但有些又是无关视频

其实我有想提这个的😂（因为我最近才开始用 `传统连播模式`，之前一个能用的只分 P 的脚本最近废掉了）

***

不过楼主想要的这个永久关闭自动连播破站自己的记忆应该就能做到的，关掉后就是分 P/合集/推荐视频全都不连播。

**Comment by timongh** [comment]
这个吧，我之前也想过加个组件来着，只不过一直拖着没动。

我的想法是把 分P连播，合集连播，推荐连播 做成 3 个独立按钮，清晰明了。再加个选项，修改仅限当次还是全局修改。

**Reply by timongh** [reply]
别对我报太高的期望，一时半会我不会动手做这个的。

有兄弟愿意动手做的话就尽管拿去吧。不过记得跟我提一嘴。

**Comment by SunJunge08** [comment]
换了台电脑，好了。
